### ISSUE TITLE
Can you help me to port a SSGI single bounce code from shader toy to unreal post process? 

### ISSUE THREAD

**Issue by JokahPokah** [Dimev/shadertoy-to-unreal-engine#2]
Hi, 
While unreal 5.1 is getting praised everywhere, when we tried to work with it, we are facing many issues such as lumen cannot be as performant as RTXGI. So, we thought that we will use SSGI instead. And then we noticed that the Epic also has "locked" the slider inside post processing volume which was controlling the intensity of SSGI lighting. 
All these made our game project awful. 

We were forced to switch back and we want to use SSR and SSGI in post processing material which is currently not accessible inside post process inputs. We want to use them for "distance lighting and reflections" with blend mask so that only nearby region of players is using RT reflections and RTXGI. There are also issues with RT reflections such as, no matter how much denoiser is applied at low sample count, anything which is rougher has noise. We want to deal with SSR with this. There is hybrid reflection option, but it doesn't provide the option to use RT reflection for <0.3 (a user selected roughness value) and then let us use SSR reflection for >0.3. 

In these days, I have searched a lot, all I see there are multiple dynamic SSGI codes and assets are available for unity which are somewhat written in HLSL code. We can buy those codes if needed and also, we came across SSGI code in ShaderToy. Now, we are small team of 3D modeling creator, texturing artists, animator. We lack skill in HLSL code, but we are very much interested into learning the HLSL to .usf and do our goal. We also check folder "Engine/Shader/Private" and we found that there are in fact SSGI and SSR .usf files with codes. So, it might also be possible to use them instead shadertoy.

We came across your work by a discord friend, as you are already familiar with the topic, can you help us get started and familiarize with the job? I can do all manual job of typing again, testing, finding modifications while you can just act as an inspector. We can also share our revenue $$$ once we make our assets open to other users and publish our game.
here is our main discord id:  IlluStudioAdmn#9942
Do let me know your feedback.

**Comment by Dimev** [comment]
I haven't used UE in a while, so I likely won't be able to help with this.
It likely also needs extra information you can't really get with materials alone, which is where my knowledge ends.

SSGI is also not great, as it doesn't have information from anything outside of the screen, so it will get strange artifacts when moving around.

If you want fast GI, then you're likely better off using Godot's SDFGI or VXGI. (Apparently the latter can run fine on iGPU's)

**Comment by JokahPokah** [comment]
As i told you before, we just need to convert it mid range.

**Comment by JokahPokah** [comment]
All i need one scenes texture 0 and one with ssgi added then i can blend it
whatever

**Comment by Dimev** [comment]
I understand what you want, but the problem is that doing SSGI is the hard part here.

**Comment by JokahPokah** [comment]
Can we have a contact through discord ? You might be able to guide me some
ureal  .usf specific terms on your casual time?

**Comment by Dimev** [comment]
No, because I barely know anything about the USF/USH stuff, and it has been a while since I used unreal.
